### ISSUE TITLE
Use Intel MKL

### ISSUE THREAD

**Issue by GoogleCodeExporter** [adda-team/adda#67]
```
Support using Intel MKL, when it is available. In particular, it may help
with linear algebra functions.
```

Original issue reported on code.google.com by `yurkin` on 7 Apr 2009 at 5:54

**Comment by myurkin** [comment]
In particular, Intel BLAS should be relevant for #61

**Comment by DaveOri** [comment]
Sorry, I can't see the point of this issue.
Intel MKL should include optimized versions of the BLAS, LAPACK, scaLAPACK and also FFT routines.
MKL should also use (as I can remember) the standard BLAS and LAPACK interfaces so the code should be written using BLAS and LAPACK calls and the user who got MKL license only needs to target MKL libraries at run time to boost his/her performance.

**Comment by myurkin** [comment]
I agree. The point of this issue is to actually try to do it and make the switch convenient, e.g. a simple option to make. Anyway, the main work is #61 and then the user may indeed bring in MKL himself.

But in terms of FFT, some coding changes will most probably be required (judging from the current code for using different FFT implementations).

Overall, there is really low priority for this changes.
